### PR TITLE
Added VA test[V2]

### PR DESCRIPTION
**Pull request by PraveenPenguin**
added VA test which test virtual address limit on linux

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
LOG:

```Command './va_test -s 1' finished with 255 after 0.00183987617493s
[stdout] Scenario 1 : Alloc 64k page from 0-512 tb VA
[stdout] 
[stdout] Allocating 64k pages < 128TB 
[stdout] Map failed address < 128TB 0xffffffffffffffff 
[stdout] in low Bad address ffffffffffffffff
[stdout] 
[stdout]  Address in > 128Tb !!! So Problem 

Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_framework-55.0-py2.7.egg/avocado/core/test.py:817
Traceback (most recent call last):
  File "/home/praveen/LE_Desktop/mywork/avacado/localavocado/avocado-misc-tests/memory/vatest.py", line 82, in test
    self.fail("test failed, Please check debug log for failed"
  File "/usr/lib/python2.7/site-packages/avocado_framework-55.0-py2.7.egg/avocado/core/test.py", line 983, in fail
    raise exceptions.TestFail(message)
TestFail: test failed, Please check debug log for failedtest cases

Local variables:
 -> line <type 'str'>:  Address in > 128Tb !!! So Problem 
 -> result <class 'avocado.utils.process.CmdResult'>: Command: ./va_test -s 1
Exit status: 255
Duration: 0.00183987617493
Stdout:

Scenario 1 : Alloc 64k page from 0-512 tb VA

Allocating 64k pages < 128TB 
Map failed address < 128TB 0xffffffffffffffff 
in low Bad address ffffffffffffffff

 Address in > 128Tb !!! So Problem 

Stderr:

PID:
8476

 -> self <class 'vatest.VATest'>: 1-vatest.py:VATest.test
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
FAIL 1-vatest.py:VATest.test -> TestFail: test failed, Please check debug log for failedtest cases

Not logging /proc/pci (file does not exist)
Not logging /proc/slabinfo (lack of permissions)
Not logging /sys/kernel/debug/sched_features (file does not exist)
Test results available in /home/praveen/avocado/job-results/job-2017-11-07T10.52-61def93
[praveen@localhost memory]$ 
```

**Comment by PraveenPenguin** [comment]
@narasimhan-v is this log is sufficient if not will attach job log file

File: memory/vatest.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            else:
+                memory.set_num_huge_pages(1024)
+        elif self.scenario_arg in [5, 6, 7]:
+            if memory.meminfo.Hugepagesize.gb != 16:
+                self.cancel(
+                    "Test need to skip as 16GB huge need to configured")
+            if self.scenario_arg == 7:
+                memory.set_num_huge_pages(2)
+            else:
+                memory.set_num_huge_pages(1)
```
Comment (narasimhan-v) [inline memory/vatest.py]: What will the test do if `scenario_arg` is a value not handled ?

Can it be like:
```
if memory.meminfo.Hugepagesize.mb != 16:
    self.cancel()
dic = {'2': 1024, '3', 1024, ...}
if self.scenario_arg in dic:
    memory.set_num_huge_pages(dic[self.scenario_arg])
```

Comment (PraveenPenguin) [reply memory/vatest.py]: yes make sense , thanks

File: memory/vatest.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    """
+
+    def setUp(self):
+        '''
+        Build VA Test
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        self.scenario_arg = self.params.get('scenario_arg', default=1)
```
Comment (narasimhan-v) [inline memory/vatest.py]: @PraveenPenguin small change. `int(self.params.get('scenario_arg', default=1))`. Except that, I am fine with the test.

File: memory/vatest.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        dic = {2: 1024, 3: 1024, 4: 131072, 5: 1, 6: 1, 7: 2}
+        if self.scenario_arg not in range(2, 12):
+            self.cancel("Test need to skip as scenario will be 1-7")
```
Comment (narasimhan-v) [inline memory/vatest.py]: Is it 1-7 or 2-12 ?

File: memory/vatest.py (hunk 2 above)
Comment (PraveenPenguin) [reply memory/vatest.py]: sure will do

File: memory/vatest.py (hunk 3 above)
Comment (PraveenPenguin) [reply memory/vatest.py]: sure , sorry for typo

**Comment by PraveenPenguin** [comment]
@narasimhan-v thanks for review .made changes can you please have a look

**Pull request merged**